src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs

```diff
@@ -117,7 +117,13 @@ bool MethodCallSelf(MethodDefinition method)
         {
             foreach (var instruction in method.Body.Instructions)
             {
+#pragma warning disable IDE0079 // Remove unnecessary suppression
```

Comment:
If you set the LangVersion explicitly I don't think you need this anymore

Reply:
That would require setting the LangVersion to 9, but we're generally avoiding going beyond 8 at this point.

Reply:
I get the benefit of suppressing for the source packages. But here I don't understand the reasoning. What is wrong by explicitly setting the LangVersion if we are deciding that we don't want to go above 8 at the state? It seems to me setting the LangVersion instead of letting the TFM decide would be the better option than decoarting the code with suppression?

> That would require setting the LangVersion to 9, but we're generally avoiding going beyond 8 at this point.

LangVersion 8 wuold have been enough for this code. See

![image](https://user-images.githubusercontent.com/174258/144188816-3e914af7-05f3-4c4d-8767-ecb1c7a539e6.png)

I'm curious about the reasoning so that we can leverage that as some coding principle somewhere so that we do it consistently across the code base. I think it is necessary to do that otherwise people will introduce the LangVersion again like we have done in 

https://github.com/Particular/NServiceBus.RavenDB/pull/801/commits/6a7ef99110afa4d2025ef805a553a17143572de2